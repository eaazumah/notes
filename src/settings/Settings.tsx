import React, {useState} from 'react';
import styled from 'styled-components/native';
import pkg from '../../package.json';
import Header from './components/Header';
import ThemePicker from './components/ThemePicker';
interface Props {}

const Settings: React.FC<Props> = () => {
  const [isVisible, setIsVisible] = useState(false);
  const showThemePicker = () => setIsVisible(true);
  return (
    <Container>
      <Header />
      <Content>
        <Item onPress={showThemePicker} testID={'theme-item'}>
          <Text>Theme</Text>
        </Item>
        <Item>
          <Text>App Version: {pkg.version}</Text>
        </Item>
      </Content>
      <ThemePicker isVisible={isVisible} setIsVisible={setIsVisible} />
    </Container>
  );
};

const Text = styled.Text`
  color: ${props => props.theme.primaryText};
`;

const Container = styled.View`
  flex: 1;
  background-color: ${props => props.theme.backgroundColor};
`;

const Content = styled.View`
  flex: 1;
  align-items: center;
  justify-content: center;
  background-color: ${props => props.theme.backgroundColor};
`;

const Item = styled.TouchableOpacity`
  width: 100%;
  height: 55px;
  align-items: center;
  flex-direction: row;
  padding-horizontal: 27px;
  border-bottom-width: 1px;
  justify-content: space-between;
  border-bottom-color: ${props => props.theme.surface};
`;

export default Settings;
