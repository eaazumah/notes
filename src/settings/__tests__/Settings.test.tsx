import {cleanup, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import TestProvider from '../../__testUtils__/Providers';
import Settings from '../Settings';

describe('Settings', () => {
  afterEach(cleanup);

  test('Settings renders correctly', () => {
    const screen = render(
      <TestProvider>
        <Settings />
      </TestProvider>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('it should update showThemePicker on theme item press', () => {
    const screen = render(
      <TestProvider>
        <Settings />
      </TestProvider>,
    );
    const picker = screen.getByTestId('theme-picker');
    expect(picker.props.isVisible).toEqual(false);

    const item = screen.getByTestId('theme-item');

    fireEvent.press(item);
    expect(picker.props.isVisible).toEqual(true);
  });
});
