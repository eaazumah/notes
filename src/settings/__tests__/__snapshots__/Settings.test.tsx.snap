// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Settings renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F5F5F5",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 50,
          "paddingHorizontal": 17,
          "width": "100%",
        },
      ]
    }
    testID="header-editor"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Icon
        color="#242A37"
        name="arrow-back"
        size={25}
        testID="icon"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#e6e6e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 55,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 27,
          "width": "100%",
        }
      }
      testID="theme-item"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#242A37",
            },
          ]
        }
      >
        Theme
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#e6e6e6",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 55,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingHorizontal": 27,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#242A37",
            },
          ]
        }
      >
        App Version: 
        0.0.1
      </Text>
    </View>
  </View>
  <Modal
    isVisible={false}
    testID="theme-picker"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5F5F5",
            "borderRadius": 6,
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5F5F5",
            "borderBottomColor": "#e6e6e6",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderBottomWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="light-item"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#242A37",
              },
            ]
          }
        >
          Light
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5F5F5",
            "borderBottomColor": "#e6e6e6",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderBottomWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="dark-item"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#242A37",
              },
            ]
          }
        >
          Dark
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5F5F5",
            "borderBottomColor": "#e6e6e6",
            "borderBottomLeftRadius": 6,
            "borderBottomRightRadius": 6,
            "borderBottomWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="system-item"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#242A37",
              },
            ]
          }
        >
          System Theme
        </Text>
      </View>
    </View>
  </Modal>
</View>
`;
